fdbserver/workloads/TenantManagementWorkload.actor.cpp

```diff
@@ -622,6 +622,51 @@ struct TenantManagementWorkload : TestWorkload {
 		}
 	}
 
+	// set a random watch on a tenant
+	ACTOR static Future<ErrorOr<Void>> watchTenant(TenantManagementWorkload* self, TenantName tenant) {
+		try {
+			state Reference<ReadYourWritesTransaction> tr(new ReadYourWritesTransaction(self->dataDb, tenant));
+			loop {
+				try {
+					state Future<Void> watch = tr->watch(doubleToTestKey(deterministicRandom()->random01()));
+					wait(tr->commit());
+					wait(watch);
+					return Void();
+				} catch (Error& e) {
+					wait(tr->onError(e));
+				}
+			}
+		} catch (Error& e) {
+			return e;
+		}
+	}
+
+	void checkWatchOnDeletedTenant(std::vector<std::pair<TenantName, Future<ErrorOr<Void>>>>& watchFutures) {
+		for (auto& [t, f] : watchFutures) {
+			if (!(f.get().isError(error_code_tenant_removed) || f.get().isError(error_code_tenant_not_found))) {
```

Comment:
We may be able to avoid the `tenant_not_found` error by first committing the watch and then proceeding with the tenant deletion.

Reply:
It's possible another client already do a deletion while this client commiting a watch

Reply:
However, it's still hard to control the order of watch is set before the deletion of a tenant.
I encountered a situation like
1. a watch is committed by transaction A
2. transaction A gets a `transaction_too_old` or `future_version` exception
3. tenant is deleted by transaction B
4. transaction A retries and commits the watch again
5. the watch gets `tenant_not_found`

Reply:
Yeah I think I see that case, and I believe committing (and waiting for the commit to succeed) before using the watch and then subsequently deleting the tenant could avoid the `tenant_not_found` error.